aria_data_loaders/aria_data_utils/aria_streamer.py

```diff
@@ -0,0 +1,842 @@
+
+        # hand_bd_body_T_odom_dict = frame_tree_snapshot_hand.child_to_parent_edge_map.get(
+        #     "odom"
+        # ).parent_tform_child
+        # hand_mn_body_T_odom = self.spot.convert_transformation_from_BD_to_magnun(
+        #     hand_bd_body_T_odom_dict
+        # )
+        # hand_mn_odom_T_body = hand_mn_body_T_odom.inverted()
+
+        # print("hand__body_T_odom", hand_mn_body_T_odom)
+        # print("hand__odom_T_body", hand_mn_odom_T_body)
+
+        hand_mn_body_T_handcam = hand_mn_body_T_wrist @ hand_mn_wrist_T_handcam
+        # hand_mn_odom_T_handcam = hand_mn_odom_T_body @ hand_mn_body_T_handcam
+        # hand_mn_odom_T_wrist = hand_mn_odom_T_body @ hand_mn_body_T_wrist
+
+        # print(f"wrist_T_handcam - mn: {hand_mn_wrist_T_handcam}")
+        # print(f"body_T_wrist - mn: {hand_mn_body_T_wrist}")
+        # print(f"body_T_handcam - mn : {hand_mn_body_T_handcam}")
+        # print(f"odom_T_handcam - mn : {hand_mn_odom_T_handcam}")
+        # print(f"odom_T_wrist - mn : {hand_mn_odom_T_wrist}")
+        return hand_mn_body_T_handcam
+
+    def _get_body_T_headcam(self, frame_tree_snapshot_head):
+        # print(frame_tree_snapshot_head)
+        head_bd_fr_T_frfe_dict = frame_tree_snapshot_head.child_to_parent_edge_map.get(
+            "frontright_fisheye"
+        ).parent_tform_child
+        head_mn_fr_T_frfe_dict = self.spot.convert_transformation_from_BD_to_magnun(
+            head_bd_fr_T_frfe_dict
+        )
+
+        head_bd_head_T_fr_dict = frame_tree_snapshot_head.child_to_parent_edge_map.get(
+            "frontright"
+        ).parent_tform_child
+        head_mn_head_T_fr = self.spot.convert_transformation_from_BD_to_magnun(
+            head_bd_head_T_fr_dict
+        )
+
+        head_bd_body_T_head_dict = (
+            frame_tree_snapshot_head.child_to_parent_edge_map.get(
+                "head"
+            ).parent_tform_child
+        )
+        head_mn_body_T_head = self.spot.convert_transformation_from_BD_to_magnun(
+            head_bd_body_T_head_dict
+        )
+
+        # head_bd_body_T_odom_dict = frame_tree_snapshot_head.child_to_parent_edge_map.get(
+        #     "odom"
+        # ).parent_tform_child
+        # head_mn_body_T_odom = self.spot.convert_transformation_from_BD_to_magnun(
+        #     head_bd_body_T_odom_dict
+        # )
+        # head_mn_odom_T_body = head_mn_body_T_odom.inverted()
+
+        # print("head__body_T_odom", head_mn_body_T_odom)
+        # print("head__odom_T_body", head_mn_odom_T_body)
+
+        head_mn_head_T_frfe = head_mn_head_T_fr @ head_mn_fr_T_frfe_dict
+        head_mn_body_T_frfe = head_mn_body_T_head @ head_mn_head_T_frfe
+        # head_mn_odom_T_frfe = head_mn_odom_T_body @ head_mn_body_T_frfe
+
+        # print(f"head__body_T_frfe - mn: {head_mn_body_T_frfe}")
+        # print(f"head__odom_T_frfe - mn : {head_mn_odom_T_frfe}").
+        return head_mn_body_T_frfe
+
+    def _get_odom_T_handcam(self, frame_tree_snapshot_hand):
```

Comment:
I totally skipped through these functions and am assuming they are all correct :P Overall, might be good to write a comment to explain what they should contain so that we can keep bugs in check esp. with all the transformations. Right now I am concerned that people outside the main team will be totally lost with these transformations and wont be able to help with spotting any bugs.

Reply:
@rutadesai  today me & @zephirefaith found a better way to do the same.

Priyam has created this task - https://github.com/facebookresearch/spot-sim2real/issues/83 to track the same